### ISSUE TITLE
Working with .Net 4.5? MissingMethodException

### ISSUE THREAD

**Issue by dagtorgersbraten** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#497]
Hi, my project is using .Net 4.5.  
I first get the messgae "Method 'SaveChangesAsync' does not have an implementation." but this is resolved by adding "Settings.TargetFrameworkVersion = 4.0f;" to the .tt file.  
However then when calling Savechanges() on the db context I get the below message at the method:
    public ClassificationConfiguration()
            : this("dbo")    { }

"An exception of type 'System.MissingMethodException' occurred in App.exe but was not handled in user code
Additional information: Method not found: 'System.Data.Entity.ModelConfiguration.Configuration.PrimitivePropertyConfiguration System.Data.Entity.ModelConfiguration.Configuration.PrimitivePropertyConfiguration.HasDatabaseGeneratedOption(System.Nullable`1<System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption>)'."

**Comment by dagtorgersbraten** [comment]
Sorry I get this on Add() not SaveChanges()

**Comment by sjh37** [comment]
Hi. Async was added in 4.5 so should be compatible with and without using Async methods.
What version of EntityFramework nuget package did you install?

**Comment by dagtorgersbraten** [comment]
Hi, thanks for speedy reply.  I tried with 6.2.0 and 6.1.3 but ran into same issue with both.

**Comment by sjh37** [comment]
Ok, that is very odd. Never seen that error before. Create a new scratch sln, and add EF, and add in the efrpg generator and see what happens from a new project pointing to your database. It might be that some of the ttinclude files have been altered or lines deleted causing you issues.

**Comment by dagtorgersbraten** [comment]
OK thanks Simon, I just created a new solution, added EF 6.1.3 and then
added efrpg v2.27.3.  I am using case sensitive SQL DB so I needed to go
into the "EF.Reverse.POCO.Core.ttinclude" and replace "C.TABLE_NAME" with
"c.TABLE_NAME".

I then just added the following and ran the app

using (var db = new MyDbContext())
But I get

"Additional information: Method 'SaveChangesAsync' in type
'BAM.MyDbContext' from assembly 'BAM, Version=1.0.0.0, Culture=neutral,
PublicKeyToken=null' does not have an implementation."

Although looking in the generated EF .cs file I can see the methods are
there...strange, not sure what is happening but I have tried a new solution
a couple of times and get the same.

public System.Threading.Tasks.Task<int> SaveChangesAsync()

{

++SaveChangesCount;

return System.Threading.Tasks.Task<int>.Factory.StartNew(() => 1);

}

**Comment by dagtorgersbraten** [comment]
Hi again, a bit more info...
I found both public/private methods for SaveChangesAsync,when I commented
out the private methods it then does not show this SaveChangesAsync error
anymore.
Here are the private ones:
System.Threading.Tasks.Task<int> SaveChangesAsync();

System.Threading.Tasks.Task<int> SaveChangesAsync(System.Threading.
CancellationToken cancellationToken);

And I also found these public ones:

public System.Threading.Tasks.Task<int> SaveChangesAsync(){

++SaveChangesCount;

return System.Threading.Tasks.Task<int>.Factory.StartNew(() => 1);}

After commenting out the private SaveChangesAsync,I then get the below
"original" message from the thread:

An unhandled exception of type 'System.MissingMethodException' occurred in
EntityFramework.dll
Additional information: Method not found:
'System.Data.Entity.ModelConfiguration.Configuration.PrimitivePropertyConfiguration
System.Data.Entity.ModelConfiguration.Configuration.PrimitivePropertyConfiguration.HasDatabaseGeneratedOption

**Comment by dagtorgersbraten** [comment]
Just an update on the previous, the reason why there were 2 SaveChangesAsync()
was becauase "AddUnitTestingDbContext" was set to True so it had a fake
DbContext setup with an implementation...

**Comment by sjh37** [comment]
Thanks. I'll check this out on Monday.

**Comment by sjh37** [comment]
Hi. I had a look this morning and created a small .Net 4.5 project.

The private `System.Threading.Tasks.Task<int> SaveChangesAsync();` you mentioned, is not private, it's on an interface so does not yet have have an access modifier.

There is no implementation of SaveChangesAsync() in the generated DbContext because it is implemented by the base class `System.Data.Entity.DbContext`

The Fake Db Context must implement it as it's base class is only `IMyDbContext`, hence it has a body of 
```c#
public System.Threading.Tasks.Task<int> SaveChangesAsync()
{
    ++SaveChangesCount;
    return System.Threading.Tasks.Task<int>.Factory.StartNew(() => 1);
}
```

I cannot find a fault with the code generated, nor running the code and calling SaveChangesAsync();
Can you check at the top of the generator `*.tt*` files there is a version comment with `v2.37.3`. Just in case you are using an old version which contained a bug.

Here is a [link](http://www.hicrest.net/code/DeleteMe4_5.zip) to the test I created for this case.

**Comment by dagtorgersbraten** [comment]
Hi Simon,
Thanks for taking the time to test that.
I downloaded the project and ammended the connection, re-saved the .tt file
and amended the program slightly to use my DB.  It all builds successfully
and looks good but when I run it I get the same error as last time:

*"Method 'SaveChangesAsync' in type 'DeleteMe4_5.MyDbContext' from assembly
'DeleteMe4_5, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does
not have an implementation."*

I think this must be somehting strange with my environment.  I have tried
this in VS2015 and VS2017.  Strangely it works perfectly with .Net 4 but
anything higher and I get the above error.

Anyway thanks for your help.
